pkg/collector/corechecks/net/wlan/wlan.go

```diff
@@ -104,42 +104,81 @@ func (c *WLANCheck) isChannelSwap(wi *wifiInfo) bool {
 	return c.lastChannel != wi.channel
 }
 
+// Status metric values (replacing deprecated service checks)
+const (
+	statusOK       float64 = 0 // WiFi operational
+	statusWarning  float64 = 1 // WiFi interface inactive
+	statusCritical float64 = 2 // WiFi collection failed
+)
+
 // Run runs the check
 func (c *WLANCheck) Run() error {
 	sender, err := c.GetSender()
 	if err != nil {
 		return err
 	}
+
+	// Attempt to get WiFi info from GUI via IPC
 	wi, err := getWiFiInfo()
 	if err != nil {
-		log.Error(err)
+		// Failed to get WiFi info - emit CRITICAL status
+		log.Errorf("WLAN check failed: %v", err)
+		log.Error("Ensure the Datadog Agent GUI is running for WiFi metrics collection on macOS 15+")
+
+		// Emit status metric: CRITICAL (replaces deprecated service check)
+		sender.Gauge("system.wlan.status", statusCritical, "", []string{
+			"status:critical",
+			"reason:ipc_failure",
+		})
+		// Track error count for monitoring
+		sender.Count("system.wlan.check.errors", 1, "", []string{
+			"error_type:ipc_failure",
+		})
+		sender.Commit()
 		return err
 	}
 
+	// Check if WiFi interface is active
 	if wi.phyMode == "None" {
 		log.Warn("No active Wi-Fi interface detected: PHYMode is none.")
+
+		// Emit status metric: WARNING (replaces deprecated service check)
+		sender.Gauge("system.wlan.status", statusWarning, "", []string{
+			"status:warning",
+			"reason:interface_inactive",
+		})
+		sender.Commit()
 		return nil
 	}
 
+	// Prepare tags
 	ssid := wi.ssid
 	if ssid == "" {
 		ssid = "unknown"
+		log.Debug("SSID is empty - this may indicate missing location permission")
 	}
 	bssid := wi.bssid
 	if bssid == "" {
 		bssid = "unknown"
+		log.Debug("BSSID is empty - this may indicate missing location permission")
 	}
 
 	macAddress := strings.ToLower(strings.ReplaceAll(wi.macAddress, " ", "_"))
 	if macAddress == "" {
 		macAddress = "unknown"
 	}
 
-	tags := []string{}
-	tags = append(tags, "ssid:"+ssid)
-	tags = append(tags, "bssid:"+bssid)
-	tags = append(tags, "mac_address:"+macAddress)
+	tags := []string{
+		"ssid:" + ssid,
+		"bssid:" + bssid,
+		"mac_address:" + macAddress,
+		"status:ok",
+	}
+
+	// WiFi data collected successfully - emit OK status (replaces deprecated service check)
+	sender.Gauge("system.wlan.status", statusOK, "", tags)
```

Comment:
You probably need to add this metric to metadata.csv file in integration-core to reflect this new metric